### PR TITLE
Adds the Tzimisce Clan as a midround/latejoin

### PR DESCRIPTION
**Pull request by SgtHunk**
## About The Pull Request

Adds the Tzimisce clan, restricted to midround/latejoin. They get higher unspent levels ( 3 - 4 compared to 2 - 3), and are forced to be Tzimisce.
The Tzimisce clan specializes in shapeshifting. Namely, they specialize in shifting flesh into a new form.
This will be reflected in their ability to turn victims into monsters of ferocious power - abominations made of flesh created to do the Tzimisce's bidding.
As for their bane? They are forced to sleep in the coffin that they claimed - making them unable to go around coffins and make multiple hidden lairs, rather forced to stay in one place.

TO DO:

- [ ] Actually add the bane (add another one if its too weak?)
- [ ] Actually add the monsters (I NEED SPRITES)
- [ ] Properly test this
- [ ] Integrate tzimisce as a prob() in the midround/latejoin bloodsucker executions [Scrapped because I want admins to at least be able to trigger it without relying on %]
- [ ] Probably add more content? (abilities? give them Veil?)
- [ ] Properly test this x2

Also here's a picture of an "Acrobat" as seen in Vtm:B
Isn't he beautiful
image

@JohnFulpWillard hi
## Why It's Good For The Game

 More content!!!!!!!!!!!!!!!!!!!!!!!!!
Also indirectly adds a minor threat to other bloodsuckers that can rarely happen in dynamic.

## Changelog
add: Added Tzimisce as a rare clan that non-roundstart bloodsuckers can get. Beware, Nosferatu!
/:cl:

### CODE REVIEW

File: fulp_modules/main_features/bloodsuckers/code/antagonists/bloodsucker_datum.dm [hunk 1]
```diff
@@ -769,6 +769,61 @@
 
 	return fullname
 
+/// Special midround/latejoin only bloodsucker - used to override procs accordingly.
+/datum/antagonist/bloodsucker/tzimisce
```
Comment (JohnFulpWillard) [inline fulp_modules/main_features/bloodsuckers/code/antagonists/bloodsucker_datum.dm]: I dont like this at all. you can do this instead

/datum/dynamic_ruleset/midround/tzimisce_bloodsucker/execute()
	var/mob/M = pick(living_players)
	assigned += M.mind
	living_players -= M.mind
	var/datum/antagonist/bloodsucker/sucker = new
	M.mind.add_antag_datum(sucker)
	sucker.my_clan = CLAN_TZIMISCE
	message_admins("[ADMIN_LOOKUPFLW(M)] was selected by the [name] ruleset and has been made into a midround Bloodsucker.")
	log_game("DYNAMIC: [key_name(M)] was selected by the [name] ruleset and has been made into a midround Bloodsucker.")
	return TRUE

From there, we can change AssignClanAndBane accordingly to fit with these changes, like not assigning a clan if you're already in one.

File: fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_flaws.dm [hunk 2]
```diff
@@ -79,6 +79,16 @@
...
 	var/mob/living/carbon/human/bloodsucker = owner.current
+	if(is_tzimisce)
+		my_clan = CLAN_TZIMISCE
+		to_chat(owner, "<span class='announce'>You are a fierce, powerful shapeshifter! You are Tzimisce!<br> \
+			* As part of the Tzimisce Clan, you are able to shapeshift your victims on a persuasion rack to turn them into monsters!<br> \
+			* Your bane has made you very vulnerable to stakes, and forced to remain in a static coffin - you are unable to sleep in unclaimed coffins.</span>")
+		/// TODO: MOVE A LOT OF STUFF HERE.
+		return
+	/// Clan already assigned? Then don't assign it again.
+	if(my_clan)
```
Comment (JohnFulpWillard) [inline fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_flaws.dm]:
```suggestion
	if(my_clan != null)
```

**Comment by JohnFulpWillard** [comment]
You can try asking AspEv for sprites?

**Comment by SgtHunk** [comment]
tbh I'll probably just save sprites for last
I want to at least get the rest of things done before asking for sprites
Don't want to use up anyone's time but my own

File: fulp_modules/main_features/bloodsuckers/code/mobs/twoarmed.dm [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	turns_per_move = 5
+	see_in_dark = 4
+	attack_vis_effect = ATTACK_EFFECT_CLAW
+	initial_language_holder = /datum/language_holder/vampiric_mob
+	damage_coeff = list(BRUTE = 0.4, BURN = 0.7, TOX = 0, CLONE = 0, STAMINA = 0, OXY = 1)
+
+/mob/living/simple_animal/hostile/retaliate/tzimisce_twoarmed/examinate(atom/A, mob/living/carbon/human/user)
+	. = ..()
+	var/datum/antagonist/bloodsucker/bloodsuckerdatum = IS_BLOODSUCKER(user)
+	if(bloodsuckerdatum.my_clan == CLAN_TZIMISCE)
```
Comment (JohnFulpWillard) [inline fulp_modules/main_features/bloodsuckers/code/mobs/twoarmed.dm]: Should non Tzimisce bloodsuckers know anything about them?

Or observers, like the persuasion rack.

File: fulp_modules/main_features/bloodsuckers/code/mobs/twoarmed.dm [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	melee_damage_upper = 20
+	melee_damage_type = BURN
+	attack_verb_continuous = list("claws", "slashes")
+	attack_verb_simple = list("claw", "slash")
+	speed = 0
+	turns_per_move = 5
+	see_in_dark = 4
+	attack_vis_effect = ATTACK_EFFECT_CLAW
+	initial_language_holder = /datum/language_holder/vampiric_mob
+	damage_coeff = list(BRUTE = 0.4, BURN = 0.7, TOX = 0, CLONE = 0, STAMINA = 0, OXY = 1)
```
Comment (JohnFulpWillard) [inline fulp_modules/main_features/bloodsuckers/code/mobs/twoarmed.dm:21]: So they're gonna be taking damage from the lack of o2?

Comment (SgtHunk) [reply fulp_modules/main_features/bloodsuckers/code/mobs/twoarmed.dm:21]: I'm going to change it - though that variable is for damage multipliers
the var that'd make them unable to take oxy would be changing everything in atmos_requirements to 0

File: fulp_modules/main_features/bloodsuckers/code/mobs/twoarmed.dm (hunk 3 above)
Comment (SgtHunk) [reply fulp_modules/main_features/bloodsuckers/code/mobs/twoarmed.dm]: Yeah, I'll make it so they're able to.

**Comment by SgtHunk** [comment]
To Fulp's git this goes so I can properly update my branch and receive a bit more feedback whilst I work on this

**Pull request closed without merging**